packages/taquito-local-forging/test/taquito-local-forging.spec.ts

```diff
@@ -19,153 +20,255 @@ describe('Forge and parse operations default protocol', () => {
+      done();
     });
 
-    test('Verify protocol Kathmandu is not inferior to protocol Babylon', () => {
-      const falseProtoInferiorTo = ProtoInferiorTo(b, a);
-      expect(falseProtoInferiorTo).toEqual(false);
+    test(`Verify getCodec for CODEC.OP_DELEGATION`, async (done) => {
+      const codec = CODEC.OP_DELEGATION
+      const myGetCodec = getCodec(codec, ProtocolsHash.PtKathman)
+      const consumer = myGetCodec.decoder(hexToParse)
+      expect(consumer).toEqual({ "counter": "161756491", "fee": "114", "gas_limit": "103", "storage_limit": "70" })
+      done();
+    });
+
+    test(`Verify Arrow Function for CODEC.OP_SEED_NONCE_REVELATION`, async (done) => {
+      const codec = CODEC.OP_SEED_NONCE_REVELATION
+      const myGetCodec = getCodec(codec, ProtocolsHash.PtKathman)
+      const gotCodec = myGetCodec.decoder(hexToParse)
+      expect(gotCodec).toStrictEqual({ "level": 91409386, "nonce": "904d67468808c8eb50a9450c9721db309128012543902d0ac358a62ae28f75bb" })
+
+      const consumer = Uint8ArrayConsumer.fromHexString(hexToParse)
+      expect(decoders[codec](consumer)).toStrictEqual(gotCodec)
+      expect((schemaDecoder(decoders)(SeedNonceRevelationSchema)(consumer))).toBeDefined
+      done();
+    });
+
+    test(`Verify Arrow Functions for CODEC.SECRET is toHexString(val.consume(20))`, async (done) => {
+      const codec = CODEC.SECRET
+      const myGetCodec = getCodec(codec, ProtocolsHash.PtKathman)
+      const encodeCodec = myGetCodec.encoder('0572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62a')
+      expect(encodeCodec).toEqual("0572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62a")
+      const decodeCodec = myGetCodec.decoder('0572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62a')
+      expect(decodeCodec).toEqual("0572cbea904d67468808c8eb50a9450c9721db30")
+      done();
+    });
+
+    test(`Verify Arrow Function for CODEC.RAW is toHexString(val.consume(32)),`, async (done) => {
+      const codec = CODEC.RAW
+      const myGetCodec = getCodec(codec, ProtocolsHash.PtKathman)
+      const encodeCodec = myGetCodec.encoder('0572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62a')
+      expect(encodeCodec).toEqual("0572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62a")
+      const decodeCodec = myGetCodec.decoder('0572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62a')
+      expect(decodeCodec).toEqual("0572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62a")
+      done();
+    });
+
+    test(`Verify Arrow Function for CODEC.OP_ACTIVATE_ACCOUNT`, async (done) => {
+      const codec = CODEC.OP_ACTIVATE_ACCOUNT
+      const myGetCodec = getCodec(codec, ProtocolsHash.PtKathman)
+      const gotCodec = myGetCodec.decoder('0572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62a')
+      expect(gotCodec).toEqual({ "pkh": "tz1L8qbHKzqFmLiZqxaJdMEWrCwmmrSw2wj4", "secret": "9128012543902d0ac358a62a" })
```

Comment:
Secret seems too short, are we missing a validation there? Do we need a follow-up ticket?

Reply:
The hex string i was encoding was not complete. It was changed to use a proper hex string. Now the secret is the correct length.